e2e-polybft/framework/test-cluster.go

```diff
@@ -0,0 +1,436 @@
+package framework
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	// envE2ETestsEnabled signal whether the e2e tests will run
+	envE2ETestsEnabled = "E2E_TESTS"
+
+	// envLogsEnabled signal whether the output of the nodes get piped to a log file
+	envLogsEnabled = "E2E_LOGS"
+
+	// envLogLevel specifies log level of each node
+	envLogLevel = "E2E_LOG_LEVEL"
+
+	// envStdoutEnabled signal whether the output of the nodes get piped to stdout
+	envStdoutEnabled = "E2E_STDOUT"
+
+	// accountPassword is the default account password
+	accountPassword = "qwerty"
+)
+
+var startTime int64
+
+func init() {
+	startTime = time.Now().UnixMilli()
+}
+
+func resolveBinary() string {
+	bin := os.Getenv("EDGE_BINARY")
+	if bin != "" {
+		return bin
+	}
+	// fallback
+	return "polygon-edge"
+}
+
+func createAccountPasswordFile(t *testing.T) (string, func()) {
+	t.Helper()
+
+	pwdFile, err := ioutil.TempFile("", "e2e-polybft")
+	require.NoError(t, err)
+
+	_, err = pwdFile.WriteString(accountPassword)
+	require.NoError(t, err)
+	require.NoError(t, pwdFile.Close())
+
+	return pwdFile.Name(), func() {
+		err = os.Remove(pwdFile.Name())
+		require.NoError(t, err)
+	}
+}
+
+type TestClusterConfig struct {
+	t *testing.T
+
+	Name              string
+	Premine           []common.Address
+	HasBridge         bool
+	NonValidatorCount int
+	WithLogs          bool
+	WithStdout        bool
+	LogsDir           string
+	TmpDir            string
+	Binary            string
+	ValidatorSetSize  uint64
+
+	logsDirOnce sync.Once
+}
+
+func (c *TestClusterConfig) Dir(name string) string {
+	return filepath.Join(c.TmpDir, name)
+}
+
+func (c *TestClusterConfig) GetStdout(name string) io.Writer {
+	writers := []io.Writer{}
+
+	if c.WithLogs {
+		c.logsDirOnce.Do(func() {
+			c.initLogsDir()
+		})
+
+		f, err := os.OpenFile(filepath.Join(c.LogsDir, name+".log"), os.O_RDWR|os.O_APPEND|os.O_CREATE, 0660)
+		if err != nil {
+			c.t.Fatal(err)
+		}
+
+		writers = append(writers, f)
+
+		c.t.Cleanup(func() {
+			err = f.Close()
+			if err != nil {
+				c.t.Logf("Failed to close file. Error: %s", err)
+			}
+		})
+	}
+
+	if c.WithStdout {
+		writers = append(writers, os.Stdout)
+	}
+
+	if len(writers) == 0 {
+		return io.Discard
+	}
+
+	return io.MultiWriter(writers...)
+}
+
+func (c *TestClusterConfig) initLogsDir() {
+	logsDir := path.Join("..", "e2e-logs", fmt.Sprintf("e2e-logs-%d", startTime), c.t.Name())
```

Comment:
There is a warning on CI and no logs for e2e tests were uploaded (https://github.com/0xPolygon/polygon-edge/actions/runs/3272891694):

> No files were found with the provided path: e2e-logs-*/. No artifacts will be uploaded.

Reply:
Good catch